src/main/kotlin/org/wycliffeassociates/otter/common/domain/resourcecontainer/ImportResourceContainer.kt

```diff
@@ -1,194 +1,156 @@
+            if (!success) {
+                throw IOException("Could not copy resource container ${dir.name} to resource container directory")
             }
         }
-        book.addAll(chapters)
-        return book
+        return destinationDirectory
     }
 
-
-    private fun constructRoot(rc: ResourceContainer): Tree {
-        val dc = rc.manifest.dublinCore
-        val slug = dc.identifier
-        val title = dc.title
-        val label = dc.type
-        val collection = Collection(0, slug, label, title, null)
-        return Tree(collection)
+    private fun makeExpandedContainer(container: ResourceContainer): Result {
+        val dublinCore = container.manifest.dublinCore
+        if (dublinCore.type == "bundle" && dublinCore.format.startsWith("text/usfm")) {
+            return if (container.expandUSFMBundle()) Result.SUCCESS else Result.INVALID_CONTENT
+        }
+        return Result.SUCCESS
     }
 
-    private fun getCategories(rc: ResourceContainer): List<Collection> {
-        val collections = arrayListOf<Collection>()
-        val config = rc.config
-        config?.let {
-            if (it is OtterResourceContainerConfig) {
-                it.extendedDublinCore?.let {
-                    for (cat in it.categories) {
-                        collections.add(
-                                Collection(cat.sort, cat.identifier, cat.identifier, cat.title, null)
-                        )
-                    }
+    private fun constructContainerTree(container: ResourceContainer): Pair<Result, Tree> {
+        val root = Tree(container.toCollection())
+        val categoryInfo = container.otterConfigCategories()
+        for (project in container.manifest.projects) {
+            var parent = root
+            for (categorySlug in project.categories) {
+                // use the `latest` RC spec to treat categories as hierarchical
+                // look for a matching category under the parent
+                val existingCategory = parent.children
+                        .map { it as? Tree }
+                        .filter { (it?.value as? Collection)?.slug == categorySlug }
+                        .firstOrNull()
+                parent = if (existingCategory != null) {
+                    existingCategory
+                } else {
+                    // category node does not yet exist
+                    val category = categoryInfo.filter { it.identifier == categorySlug }.firstOrNull() ?: continue
+                    val categoryNode = Tree(category.toCollection())
+                    parent.addChild(categoryNode)
+                    categoryNode
                 }
             }
+            val projectResult = constructProjectTree(container.dir, project)
+            if (projectResult.first == Result.SUCCESS)
+                parent.addChild(projectResult.second)
+            else
+                return Pair(projectResult.first, Tree(Unit))
```

Comment:
use { } around the if and else since both the if and else statement here cannot exist on the same line